### PR TITLE
Upgrade SourceIndexServer to ASP.NET Core 2.0.3

### LINKED ISSUE #87: Need to update the website to ASP.NET Core 2.0
I was wondering if @alexperovich or @natemcmaster or anyone else would be interested in updating to ASP.NET Core 2.0? 

It would be much appreciated...

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves #87

### CODE REVIEW

**Comment by natemcmaster** [comment]
@KirillOsenkov it didn't require many code changes. Tests pass in VS. Is there some other validation I should check to make sure this works fine?

File: src/SourceIndexServer/Program.cs [hunk 1]
```diff
@@ -7,15 +7,12 @@ public class Program
     {
         public static void Main(string[] args)
         {
-            var host = new WebHostBuilder()
-                .UseKestrel()
-                .UseContentRoot(Directory.GetCurrentDirectory())
```
Comment (KirillOsenkov) [inline src/SourceIndexServer/Program.cs]: is this no longer needed? I think ContentRoot should still be needed so we can serve static content from the current directory?

**Comment by KirillOsenkov** [comment]
Nah, let's get this in and fix it up afterwards if something's broken :)

Cc @alexperovich if you could check this out whenever you have time?

**Comment by KirillOsenkov** [comment]
@natemcmaster I do have one question. Here are the steps I use:

1. Set HtmlGenerator as startup and F5. It will generate a website for TestSolution under `C:\SourceBrowser\src\HtmlGenerator\bin\Debug\net46\Index`
2. Now I want to set SourceIndexServer as startup and F5. The problem is I don't know how to point it to serve from the root specified in step 1. Basically how do I change the Environment.ContentRootPath from `C:\SourceBrowser\src\SourceIndexServer` to `C:\SourceBrowser\src\HtmlGenerator\bin\Debug\net46\Index`?

If all works well, if you serve the website from that page, it should show up, searching should work and clicking on search results should display the source code in the right pane.

Thanks for your help!

**Comment by KirillOsenkov** [comment]
It seems to work the other way: if you make sure HtmlGenerator generates its output to SourceIndexServer\Index, then it's all working fine.

File: src/SourceIndexServer/Program.cs (hunk 1 above)
Comment (natemcmaster) [reply src/SourceIndexServer/Program.cs]: This is part of the new default WebHost builder. Reduces the boilerplate APIs you need to call. That's also why I could remove the ILoggerFactory.AddConsole() call.

https://github.com/aspnet/MetaPackages/blob/b40a1a5a8cfb186d41689f3c319135e518a7630c/src/Microsoft.AspNetCore/WebHost.cs#L152

**Comment by natemcmaster** [comment]
> Basically how do I change the Environment.ContentRootPath 

You can still add a call UseContentRoot.

```c#
        public static IWebHost BuildWebHost(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseContentRoot(@"C:\SourceBrowser\src\HtmlGenerator\bin\Debug\net46\Index")
                .UseStartup<Startup>()
                .Build();
```

**Comment by KirillOsenkov** [comment]
I see, cool. It's interesting that if I set the Working Directory to that path in Project Properties -> Debug, the current directory at runtime doesn't seem to change. How is Environment.ContentRootPath initialized?

**Comment by natemcmaster** [comment]
Not sure what that UI changes does. Does it set `RunWorkingDirectory`? By default Microsoft.NET.Sdk.Web always runs your project from the project folder. 

https://github.com/aspnet/websdk/blob/5ab771ee5333ffa83fd7521ff6308ef863c04594/src/ProjectSystem/Microsoft.NET.Sdk.Web.ProjectSystem.Targets/netstandard1.0/Microsoft.NET.Sdk.Web.ProjectSystem.targets#L22

**Comment by KirillOsenkov** [comment]
Oh, RunWorkingDirectory? Excellent, this is what I want. Thanks!

**Pull request merged**